go/api/response_types.go

```diff
@@ -46,3 +46,87 @@ func CreateBoolResult(boolVal bool) Result[bool] {
 func CreateNilBoolResult() Result[bool] {
 	return Result[bool]{val: false, isNil: true}
 }
+
+// Enum to distinguish value types stored in `ClusterValue`
+type ValueType int
+
+const (
+	SingleValue ValueType = 1
+	MultiValue  ValueType = 2
+)
+
+// Enum-like structure which stores either a single-node response or multi-node response.
+// Multi-node response stored in a map, where keys are hostnames or "<ip>:<port>" strings.
+//
+// For example:
+//
+//	// Command failed:
+//	value, err := clusterClient.CustomCommand(args)
+//	value.IsEmpty(): true
+//	err != nil: true
+//
+//	// Command returns response from multiple nodes:
+//	value, _ := clusterClient.info()
+//	node, nodeResponse := range value.Value().(map[string]interface{}) {
+//	    response := nodeResponse.(string)
+//	    // `node` stores cluster node IP/hostname, `response` stores the command output from that node
+//	}
+//
+//	// Command returns a response from single node:
+//	value, _ := clusterClient.infoWithRoute(Random{})
+//	response := value.Value().(string)
+//	// `response` stores the command output from a cluster node
+type ClusterValue[T any] struct {
```

Comment:
This approach also looks good. Just a suggestion to keep `single` and `multi` response separate for better readability.